addons/web/static/src/webclient/actions/action_service.js

```diff
@@ -508,11 +508,16 @@ export function makeActionManager(env, router = _router) {
                     type: "ir.actions.act_window",
                     views: [[state.view_id ? state.view_id : false, "form"]],
                 };
-            } else if (state.view_type) {
+            } else {
                 // This is a window action on a multi-record view => restores it from
                 // the session storage
                 const storedAction = browser.sessionStorage.getItem("current_action");
                 const lastAction = JSON.parse(storedAction || "{}");
+                if (!state.view_type) {
+                    // If the view_type is not specified in the state (url)
+                    // use the default view_type of the current action
+                    state.view_type = lastAction.views[0][1];
```

Comment:
This will raise an error if you copy/paste the URL on another tab, or give to somebody else.
Because lastAction will be an empty Object (and lastAction.views will be undefined)

You can avoid the error with an optional chaining, but it will continue to work as now, it will remove the model, and show the previous action. This means that you will have two different behaviors, one when reloading, and another when copy/past the URL, not sure that it's what we want.

```suggestion
                    state.view_type = lastAction.views?.[0]?.[1];
```